src/openms/include/OpenMS/FORMAT/HANDLERS/MzMLHandler.h

```diff
@@ -574,42 +649,12 @@ namespace OpenMS
             tmp.setMZ(mz);
             spectrum.push_back(tmp);
 
-            //add meta data
-            UInt meta_float_array_index = 0;
-            UInt meta_int_array_index = 0;
-            UInt meta_string_array_index = 0;
-            for (Size i = 0; i < input_data.size(); i++) //loop over all binary data arrays
+            // Only if there are more than 2 data arrays, we need to check
```

Comment:
this is how I interpreted the standard (see http://www.peptideatlas.org/tmp/mzML1.1.0.html#binaryDataArrayList )

```
Data point arrays for default data arrays (m/z, intensity, time) and meta data arrays. Default data arrays MUST not have the attributes 'arrayLength' and 'dataProcessingRef'.
```

it says very clear which ones are the default arrays and there are even specific wording in the standard regarding the default arrays. The standard arrays are at least implied to be m/z vs Intensity and time vs Intensity for spectra and chromatogams. Unless somebody decides to write a spectrum and put m/z and time in there, I think we are fine